test/snippets.cpp

```diff
@@ -87,6 +87,129 @@ usingStrings()
 
 //----------------------------------------------------------
 
+void
+usingInitLists()
+{
+    {
+        //[snippet_init_list_1
+
+        value jv = 
+        {
+            { "name", "John Doe" },
+            { "active", true },
+            { "total-balance", 330.00 },
+            {
+                "account-balances", { 84, 120, 126 }
+            }
+        };
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_2
+
+        value array = { true, 2, "hello" };
+
+        // Resulting JSON:
+        // "[true, 2, "hello"]"
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_3
+
+        value nested_array = { true, 2, "hello", { "bye", 0.1, false } };
+
+        // Resulting JSON:
+        // "[true, 2, "hello", ["bye", 0.1, false]]"
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_4
+
+        value ambigious = { { "hello", 42 }, { "world", 43 } }; // Should this be an array, or an object?
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_5
+
+        value object = { { "hello", 42 }, { "world", 43 } };
+
+        // Resulting JSON:
+        // "{"hello": 42, "world": 43}"
+
+        // All the following are arrays
+
+        value jv1 = { { "make", "Tesla" }, { "model", 3 }, "black" };
+
+        value jv2 = { { "library", "JSON" }, { "Boost", "C++", "Fast", "JSON" } };
+
+        value jv3 = { { "color", "blue" }, { 1, "red" } };
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_6
+
+        value arr = { { "hello", 42 }, array{ "world", 43 } };
+
+        // Resulting JSON:
+        // "[["hello", 42], ["world", 43]]"
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_7
+
+        value obj = { { "mercury", 0.39 }, { "venus", 0.72 }, { "earth", 1 } };
+
+        // Resulting JSON:
+        // "{"mercury": 0.39, "venus": 0.72, "earth": 1}"
+
+        array arr = { { "mercury", 0.39 }, { "venus", 0.72 }, { "earth", 1 } };
+
+        // Resulting JSON:
+        // "[["mercury", 0.39], ["venus", 0.72], ["earth", 1]]"
+
+        //]
+    }
+    
+    {
+        //[snippet_init_list_8
+        
+        object obj = { { "mercury", { "distance", 0.39 } }, { "venus", { 0.72, "67 million miles" } }, { "earth", 1 } };
+
+        // Resulting JSON:
+        // "{"mercury": {"distance": 0.39}, "venus": [0.72, "67 million miles"], "earth": 1}"
+
+        //]
+    }
+
+    {
+        //[snippet_init_list_9
+        
+        object obj = { { "john", 100 }, { "dave", 500 }, { "joe", 300 } };
+        value val = { "clients", std::move(obj) };
```

Comment:
These snippets are compiled with relaxed warnings? i.e. You don't need `(void)val;` and `(void)arr;` etc?

Reply:
the `(void)val` can be placed outside the snippet if needed

Reply:
Applied.